### PR TITLE
create sphinx documentation for python samples + gdal2xyz docs

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?

python samples + gdal2xyz docs added

## What are related issues/pull requests?

https://github.com/OSGeo/gdal/pull/3416

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by idanmiara** [comment]
As this PR depends on https://github.com/OSGeo/gdal/pull/3416, I've decided to merge it into it.

**Comment by rouault** [comment]
> gdalinfo.py, ogr2ogr.py, ogrinfo.py - move from osgeo_utils/samples to osgeo_utils, as they are not samples utilities

I'm not a fan of this. We should really ere on the side of not proposing too many official utilities, especially when they are redundant with C++ ones. Those scripts were made as an illustration of how to use the Python API. They don't bring more than their C++ equivalents, and are actually lagging behind.

**Review by rouault — CHANGES_REQUESTED** [review]
See above comment

**Comment by idanmiara** [comment]
> > gdalinfo.py, ogr2ogr.py, ogrinfo.py - move from osgeo_utils/samples to osgeo_utils, as they are not samples utilities
> 
> I'm not a fan of this. We should really ere on the side of not proposing too many official utilities, especially when they are redundant with C++ ones. Those scripts were made as an illustration of how to use the Python API. They don't bring more than their C++ equivalents, and are actually lagging behind.

I see your point. I've removed that commit.

File: gdal/doc/source/api/index.rst [hunk 1]
```diff
@@ -85,6 +85,7 @@ API
 
        python_api_ref
        python_gotchas
+       python_samples
```
Comment (rouault) [inline gdal/doc/source/api/index.rst:88]: Note unrelated to that PR. The doc structure is a bit messy regarding python. We have that section, and if we go a bit further in the page, we also have a dedicated page for python. I'm not sure how, but I feel like this should be unified as the current setup is confusing.

Comment (idanmiara) [reply gdal/doc/source/api/index.rst:88]: like that https://github.com/OSGeo/gdal/pull/3687 ?

**Pull request merged**